### PR TITLE
Add MFA login flow support for cloud component

### PR DESCRIPTION
**Pull request by klejejs**
## Proposed change

This PR adds support for MFA login flow in the cloud component.

I wanted to make the TOTP code be auto-fillable by adding the `autocomplete="one-time-code"` property to the text field, but with my tests using 1Password, the auto-complete suggestion menu was rendered below the popup, likely due to the use of the HTML dialog by the underlying library, so I was unsure if that should be enabled if not working as expected.

- Requires the backend PR: https://github.com/home-assistant/core/pull/132497

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
This also needs to be implemented in the voice-wizard, where there is also a cloud login form

File: src/dialogs/voice-assistant-setup/cloud/cloud-step-signin.ts [hunk 1]
```diff
@@ -106,12 +109,30 @@ export class CloudStepSignin extends LitElement {
...
       } catch (err: any) {
         const errCode = err && err.body && err.body.code;
 
+        if (errCode === "mfarequired") {
+          const totpCode = await showPromptDialog(this, {
+            title: "Two-factor Authentication",
+            inputLabel: "TOTP code",
+            inputType: "text",
+            defaultValue: "",
+            confirmText: "Submit",
```
Comment (bramkragten) [inline src/dialogs/voice-assistant-setup/cloud/cloud-step-signin.ts:128]: Please add translations for these lines

File: src/panels/config/cloud/login/cloud-login.ts [hunk 2]
```diff
@@ -252,6 +257,19 @@ export class CloudLogin extends LitElement {
         }
       } catch (err: any) {
         const errCode = err && err.body && err.body.code;
+        if (errCode === "mfarequired") {
+          const totpCode = await showPromptDialog(this, {
+            title: "Two-factor Authentication",
+            inputLabel: "TOTP code",
+            inputType: "text",
+            defaultValue: "",
+            confirmText: "Submit",
```
Comment (bramkragten) [inline src/panels/config/cloud/login/cloud-login.ts:266]: Please add translations for these lines

**Review by bramkragten — DISMISSED** [review]

**Comment by klejejs** [comment]
@bramkragten, the backend PR has been merged now.

**Review by silamon — APPROVED** [review]

**Pull request merged**